gammapy/modeling/models/spectral.py

```diff
@@ -213,6 +213,29 @@ def evaluate_error(self, energy, epsilon=1e-4):
         """
         return self._propagate_error(epsilon=epsilon, fct=self, energy=energy)
 
+    def pivot_energy(self, energy):
+        """Return the pivot energy for a given spectral model.
+        The pivot energy is the narrowest point of the butterfly.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            The energy to evaluate the spectral model and error.
+
+        """
+        dnde, dnde_error = self.evaluate_error(energy)
+
+        y_lo = dnde - dnde_error
+        y_hi = dnde + dnde_error
+
+        # Add here to get rid of any negative fluxes
```

Comment:
Is there a way to test that in order to be sure that it works well?

Reply:
I compared values from the analytical equation in the `PowerLawSpectralModel` to this new method, and was achieving the same pivot energy